core/lib/spree/core/engine.rb

```diff
@@ -94,6 +94,14 @@ class Engine < ::Rails::Engine
           end
         end
       end
+
+      if defined?(FactoryBotRails)
```

Comment:
I don't understand why we can't use the default way of setting factories for an Engine as described [here](https://github.com/thoughtbot/factory_bot_rails#automatic-factory-definition-loading)? Is this approach excluded for a reason?

Reply:
If you're referring to the use of the Rails-style `config.factory_bot.definition_file_paths` that's not available for FactoryBot v4 (don't remember it it was introduced in 5 or 6), by updating directly `FactoryBot.definition_file_paths` we're compatible with all versions in the 4..6 range.

The `defined?` check is necessary because FactoryBot might not be available in some environments (e.g. production) and the check is done on `FactoryBotRails` with the assumption that if you're not using it, you probably also have some custom setup we don't want to mess with. In this way we can rely on the initializer as hook to inject the Solidus paths.

About using an initializer with an `after:` condition, the main reason for this is that custom factories may depend on the solidus factories, so we need a way to enforce loading them before the custom ones. Specifically `FactoryBotRails` has an initializer named `"factory_bot.set_factory_paths"` that will reset the paths, so if we add the solidus paths before that they'll be wiped out by this initializer.

Reply:
Ok, it makes sense now, thanks! So, next question: why still draft?

Reply:
I was waiting for the CI to be green, but when I pushed last week master was temporarily red, so time passed by 😉 
I just fixed the linting suggestion and repushed, will mark it as ready in a moment 👍